### ISSUE TITLE
Running tests without database causes error

### ISSUE THREAD

**Issue by YashoSharma** [HackIllinois/api-2017#50]
When running our tests on Travis-CI the database pool is destroyed causing an error to be thrown

```
Knex:warning - Pool2 - Error: Pool was destroyed
Knex:Error Pool2 - Error: ER_ACCESS_DENIED_ERROR: Access denied for user ''@'localhost' (using password: NO)
Unhandled rejection Error: Pool was destroyed
    at Pool._destroyPool (/home/travis/build/HackIllinois/api-2017/node_modules/knex/node_modules/pool2/lib/pool.js:560:20)
    at Pool.<anonymous> (/home/travis/build/HackIllinois/api-2017/node_modules/knex/node_modules/pool2/lib/pool.js:474:18)
    at Pool.<anonymous> (/home/travis/build/HackIllinois/api-2017/node_modules/knex/node_modules/pool2/lib/pool.js:517:17)
    at tryCatcher (/home/travis/build/HackIllinois/api-2017/node_modules/bluebird/js/release/util.js:16:23)
    at Promise.errorAdapter [as _rejectionHandler0] (/home/travis/build/HackIllinois/api-2017/node_modules/bluebird/js/release/nodeify.js:35:34)
    at Promise._settlePromise (/home/travis/build/HackIllinois/api-2017/node_modules/bluebird/js/release/promise.js:564:21)
    at Promise._settlePromise0 (/home/travis/build/HackIllinois/api-2017/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/home/travis/build/HackIllinois/api-2017/node_modules/bluebird/js/release/promise.js:687:18)
    at Async._drainQueue (/home/travis/build/HackIllinois/api-2017/node_modules/bluebird/js/release/async.js:138:16)
    at Async._drainQueues (/home/travis/build/HackIllinois/api-2017/node_modules/bluebird/js/release/async.js:148:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/home/travis/build/HackIllinois/api-2017/node_modules/bluebird/js/release/async.js:17:14)
    at processImmediate [as _immediateCallback] (timers.js:383:17)
Unhandled rejection Error: Pool is destroyed
    at Pool.acquire (/home/travis/build/HackIllinois/api-2017/node_modules/knex/node_modules/pool2/lib/pool.js:162:12)
    at /home/travis/build/HackIllinois/api-2017/node_modules/knex/lib/client.js:268:29
    at Promise._execute (/home/travis/build/HackIllinois/api-2017/node_modules/bluebird/js/release/debuggability.js:299:9)
    at Promise._resolveFromExecutor (/home/travis/build/HackIllinois/api-2017/node_modules/bluebird/js/release/promise.js:481:18)
    at new Promise (/home/travis/build/HackIllinois/api-2017/node_modules/bluebird/js/release/promise.js:77:14)
    at Client.acquireConnection (/home/travis/build/HackIllinois/api-2017/node_modules/knex/lib/client.js:264:21)
    at /home/travis/build/HackIllinois/api-2017/node_modules/knex/lib/runner.js:188:47
    at Promise._execute (/home/travis/build/HackIllinois/api-2017/node_modules/bluebird/js/release/debuggability.js:299:9)
    at Promise._resolveFromExecutor (/home/travis/build/HackIllinois/api-2017/node_modules/bluebird/js/release/promise.js:481:18)
    at new Promise (/home/travis/build/HackIllinois/api-2017/node_modules/bluebird/js/release/promise.js:77:14)
    at /home/travis/build/HackIllinois/api-2017/node_modules/knex/lib/runner.js:187:35
    at tryCatcher (/home/travis/build/HackIllinois/api-2017/node_modules/bluebird/js/release/util.js:16:23)
    at Function.Promise.attempt.Promise.try (/home/travis/build/HackIllinois/api-2017/node_modules/bluebird/js/release/method.js:39:29)
    at Runner.ensureConnection (/home/travis/build/HackIllinois/api-2017/node_modules/knex/lib/runner.js:186:39)
    at Runner.run (/home/travis/build/HackIllinois/api-2017/node_modules/knex/lib/runner.js:41:44)
    at QueryBuilder.Target.then (/home/travis/build/HackIllinois/api-2017/node_modules/knex/lib/interface.js:32:43)
    at QueryBuilder.tryCatcher (/home/travis/build/HackIllinois/api-2017/node_modules/bookshelf/node_modules/bluebird/js/main/util.js:26:23)
    at doThenable (/home/travis/build/HackIllinois/api-2017/node_modules/bookshelf/node_modules/bluebird/js/main/thenables.js:52:38)
    at tryConvertToPromise (/home/travis/build/HackIllinois/api-2017/node_modules/bookshelf/node_modules/bluebird/js/main/thenables.js:30:20)
    at Promise._resolveCallback (/home/travis/build/HackIllinois/api-2017/node_modules/bookshelf/node_modules/bluebird/js/main/promise.js:445:24)
    at Promise._resolveFromSyncValue (/home/travis/build/HackIllinois/api-2017/node_modules/bookshelf/node_modules/bluebird/js/main/method.js:41:14)
    at Sync.insert (/home/travis/build/HackIllinois/api-2017/node_modules/bookshelf/node_modules/bluebird/js/main/method.js:17:13)
    at null.<anonymous> (/home/travis/build/HackIllinois/api-2017/node_modules/bookshelf/lib/model.js:1009:36)
    at tryCatcher (/home/travis/build/HackIllinois/api-2017/node_modules/bookshelf/node_modules/bluebird/js/main/util.js:26:23)
    at Promise._settlePromiseFromHandler (/home/travis/build/HackIllinois/api-2017/node_modules/bookshelf/node_modules/bluebird/js/main/promise.js:510:31)
    at Promise._settlePromiseAt (/home/travis/build/HackIllinois/api-2017/node_modules/bookshelf/node_modules/bluebird/js/main/promise.js:584:18)
    at Async._drainQueue (/home/travis/build/HackIllinois/api-2017/node_modules/bookshelf/node_modules/bluebird/js/main/async.js:128:12)
    at Async._drainQueues (/home/travis/build/HackIllinois/api-2017/node_modules/bookshelf/node_modules/bluebird/js/main/async.js:133:10)
    at Immediate.Async.drainQueues [as _onImmediate] (/home/travis/build/HackIllinois/api-2017/node_modules/bookshelf/node_modules/bluebird/js/main/async.js:15:14)
    at processImmediate [as _immediateCallback] (timers.js:383:17)
  23 passing (3s)
The command "npm test" exited with 0.
```

Maybe the db needs to be mocked in the tests?
@michaelcolomb @nmagerko

**Comment by nmagerko** [comment]
Yeah the reason this is happening is because `database.js` is required in many places throughout the API. The problem with that is, during testing, we don't expect a datastore, but `database.js` still tries to connect to one.

The fix is either (as you said) to mock the datastore in some way or to find a way to stop `knex` from actually trying to connect to the datastore when `config.isTest` is true.

**Comment by nmagerko** [comment]
Fixed by #51
